### PR TITLE
Add Support for Downloading Shared Zenaroe FFmpeg Builds

### LINKED ISSUE #217: Add ability to download Shared version of FFmpeg.
To save a lot of disk space.

### PR DESCRIPTION
**Pull request by 06needhamt**
Closes #217 

This PR adds support for downloading Shared Zenaroe FFmpeg builds.

Note: I don't see any tests for the current full FFmpeg Downloader. I can look at adding some if required

### CODE REVIEW

**Comment by tomaszzmuda** [comment]
Great work :)
There are class DownloaderTests.
You missed to use it somewhere. Please add it FFmpeg.GetLatestVersion method. Create override of this method with enum as parameter with description of all parameters and mark base method as obsolete.

File: Xabe.FFmpeg/Downloader/Zeranoe/SharedFullFFmpegDownloader.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Xabe.FFmpeg.Downloader.Zeranoe
+{
+    ///<summary>Download a Full Shared build from zeranoe.com for Windows or macOS</summary>
+    internal class SharedFullFFmpegDownloader : FFmpegDownloaderBase
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Downloader/Zeranoe/SharedFullFFmpegDownloader.cs]: Change it to SharedFFmpegDownloader

Comment (06needhamt) [reply Xabe.FFmpeg/Downloader/Zeranoe/SharedFullFFmpegDownloader.cs]: Done

File: Xabe.FFmpeg/Downloader/Zeranoe/SharedFFMpegDownloader.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Xabe.FFmpeg.Downloader.Zeranoe
+{
+    ///<summary>Download a Full Shared build from zeranoe.com for Windows or macOS</summary>
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/Downloader/Zeranoe/SharedFFMpegDownloader.cs]: FFmpeg not FFMpeg

Comment (06needhamt) [reply Xabe.FFmpeg/Downloader/Zeranoe/SharedFFMpegDownloader.cs]: Done

**Comment by tomaszzmuda** [comment]
Still one think is missing - user cannot use this class.
Look at the method FFmpeg.GetLatestVersion

**Comment by 06needhamt** [comment]
Ah. got it I've added it now

File: Xabe.FFmpeg/FFmpeg.cs [hunk 3]
```diff
@@ -72,6 +74,30 @@ public static Task GetLatestVersion(bool fullVersion = false)
                 downloader = new FFmpegDownloader();
             }
             return downloader.GetLatestVersion();
+        }
+
+        /// <summary>
+        ///     Download latest FFmpeg version for current operating system to FFmpeg.ExecutablePath. If it is not set download to ".".
+        /// <param id="version">Determine which version of FFmpeg should be downloaded</param>
+        /// </summary>
+        public static Task GetLatestVersion(FFmpegVersions version)
```
Comment (tomaszzmuda) [inline Xabe.FFmpeg/FFmpeg.cs]: FFmpegVersions change this to FFmpegVersion and I will merge that :)

Comment (06needhamt) [reply Xabe.FFmpeg/FFmpeg.cs]: Done

**Pull request merged**